alica_engine/include/engine/blackboard/BlackboardBlueprint.h

```diff
@@ -17,10 +21,12 @@ class BlackboardBlueprint
     {
         vals.emplace(std::piecewise_construct, std::forward_as_tuple(key), std::forward_as_tuple(std::forward<decltype(args)>(args)...));
     }
+    void setBlackboardNode(const YAML::Node& node) { this->node = node; }
     friend class Blackboard;
 
 private:
-    std::unordered_map<std::string, std::any> vals;
+    std::unordered_map<std::string, BlackboardValueType> vals;
+    YAML::Node node;
```

Comment:
The blueprint can store the object representation of the pml file (for blackboard section) i.e. an unordered_map of <key, type_name, std::optional<default_value>>. This way the YAML::Node need not be accessed/parsed by all internals of the engine